gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/enumerated/RasOffload.scala

```diff
@@ -118,21 +119,29 @@ object RasOffload {
             // 4. Validate current node. If passed, offload it.
             validator.validate(from) match {
               case Validator.Passed =>
-                val offloadedPlan = base.offload(from)
-                val offloadedNodes = offloadedPlan.collect[ValidatablePlan] {
-                  case t: ValidatablePlan => t
-                }
-                val outComes = offloadedNodes.map(_.doValidate()).filter(!_.ok())
-                if (outComes.nonEmpty) {
-                  // 5. If native validation fails on at least one of the offloaded nodes, return
-                  // the original one.
-                  //
-                  // TODO: Tag the original plan with fallback reason. This is a non-trivial work
-                  //  in RAS as the query plan we got here may be a copy so may not propagate tags
-                  //  to original plan.
-                  from
-                } else {
-                  offloadedPlan
+                try {
+                  val offloadedPlan = base.offload(from)
+                  val offloadedNodes = offloadedPlan.collect[ValidatablePlan] {
+                    case t: ValidatablePlan => t
+                  }
+                  val outComes = offloadedNodes.map(_.doValidate()).filter(!_.ok())
+                  if (outComes.nonEmpty) {
+                    // 5. If native validation fails on at least one of the offloaded nodes, return
+                    // the original one.
+                    //
+                    // TODO: Tag the original plan with fallback reason. This is a non-trivial work
+                    //  in RAS as the query plan we got here may be a copy so may not propagate tags
+                    //  to original plan.
+                    from
+                  } else {
+                    offloadedPlan
+                  }
+                } catch {
+                  case _: GlutenNotSupportException =>
```

Comment:
Keep and disable RAS consistent.

Reply:
Would you help check the corresponding logic in legacy planner to see whether we should only catch exceptions from `doValidate()` or catch more code like this way? Thanks!

Reply:
I didn't find the place where GlutenNotSupportException is thrown in the offload rule. `doValidate()` contains `doValidateInternal()` and `doSchemaValidate()`, `doValidateInternal()` has been caught, `doSchemaValidate()` will not trigger an exception, so restore it？

Reply:
I see. So in this PR we can shrink the range of the catch block to only catch `doValidate()` to match legacy planner. Perhaps in future we should make all exceptions be caught in `doValidate` internally to make `doValidate` exception-free.